ci/infra/testrunner/README.md

```diff
@@ -1,63 +1,213 @@
-
 # Testrunner
 
-## Local Dev Machine Setup
-Set the following environment variables
+## Contents
+
+- [Summary](#summary)
+- [Configuration](#configuration-parameters)
+  - [Work Environment](#work-environment)
+  - [Platform](#platform)
+    - [Terraform](#terraform)
+    - [Openstack](#openstack)
+    - [VMware](#vmware)
+  - [Skuba](#skuba)
+  - [Log](#log)
+  - [Test](#test)
+- [Environment setup](#environment-setup)
+  - [Local setup](#local-setup)
+  - [CI setup](#ci-setup)
+- [Usage](#usage)
+  - [General CLI options](#general-cli-options)
+  - [Provision](#provision)
+  - [Node commands](#node-commands)
+    - [Node Upgrade](#node-upgrade)
+  - [Ssh](#ssh)
+  - [Test](#ssh)
+- [Examples](#examples)
+  - [Create K8s Cluster](#create-k8s-cluster)
+  - [Collect logs](#collect-logs)
+
+## Summary
+
+Testrunner is a CLI tool for setting up an environment for running e2e tests, making transparent the mechanism used for providing the test infrastructure. It can used as a stand-alone tool, running either locally or as part of a CI pipeline. It provides commands for provisioning the infrastructure, deploying a k8s cluster using `skuba` and running tests. It also provides [a library](tests/README.md) for developing `pytest`-based tests.
+
+## Configuration parameters
+
+Testrunner provides configuration by means of:
+ 
+- A yaml configuration file (defaults to `vars.yaml` in current directory)
+- Environment variables that override the configuration. Every configuration option of the form `<section>.<variable>' can be subtituted by an environment variable `SECTION_VARIABLE. Notice that some variables are defined in the "root" section (e.g `workspace`). For example `skuba.binpath` is overriden by `SKUBA_BINPATH` and `workspace` by `WORKSPACE`
+- CLI options which override some configuration parameters
+
+The following sections document the configuration options. The CLI arguments are described in the [Usage section](#usage).
+
+### Work Environment
+
+This section configures the working environment and is generally specific of each user of CI job-
+
+- workspace: path to the testrunner's working directory 
+- username: userid for the user. It's optional. If `platform.stack_name` is not set, `username` is used. 
 
 ```
-WORKSPACE="/path/to/your/workspace" # e.g. $HOME/testrunner_workspace it is expected that the skuba code is at $HOME/testrunner_workspace/skuba
-TERRAFORM_STACK_NAME="name-of-your-stack" # i.e. $USER
+workspace "/path/to/your/workspace" 
+username: "mysuserid"
```

Comment:
`myuserid` ? or maybe `USERNAME` so it is extra obvious

Reply:
USERNAME is an environment variable. I wanted to avoid confusion.